xtras/src/supervisor.rs

```diff
@@ -124,128 +120,80 @@ where
             context,
             ctor: Box::new(ctor),
             restart_policy,
-            _actor: address.clone(),
             metrics: Metrics::default(),
         };
 
         (supervisor, address)
     }
 
-    fn spawn_new(&mut self, ctx: &mut Context<Self>) {
-        let actor_name = T::name();
-        tracing::info!(actor = %&actor_name, "Spawning new actor instance");
-
-        let this = ctx.address().expect("we are alive");
-        let actor = (self.ctor)();
-
-        self.metrics.num_spawns += 1;
-        tokio_extras::spawn(&this.clone(), {
-            let task = self.context.attach(actor);
-
-            async move {
-                match AssertUnwindSafe(task).catch_unwind().await {
-                    Ok(reason) => {
-                        let _ = this
-                            .send(Stopped {
-                                reason: reason.into(),
-                            })
-                            .await;
-                    }
-                    Err(error) => {
-                        let _ = this.send(Panicked { error }).await;
-                    }
+    pub async fn run(mut self) -> (anyhow::Error, Metrics) {
```

Comment:
interesing way of returning the metrics, instead of relying on sending a message!

if you don't want to rely on the destructuring tuples, an alternative could be just returning a`Summary` struct (otherwise it's a bit odd to return `anyhow::Error` in a "positive" sense)
```rust
struct Summary {
     actor_stop_reason: anyhow::Error, 
     metrics: Metrics,
}
```

Reply:
I guess the main thing with the `anyhow::Error` is what you call it - would it be okay to just destructure it as `exit` instead of `err` when taking output from `run`?

Reply:
with the new method that hides it anyway, yes, it's OK 👍